for_curators/sample_local_dataset_upload_dev.py.ipynb

```diff
@@ -0,0 +1,305 @@
+   "source": [
+    "s3_credentials_url = f\"https://api.cellxgene.dev.single-cell.czi.technology/curation/v1/collections/{collection_id}/datasets/s3-upload-credentials\"\n",
+    "s3_cred_headers = {\"Authorization\": f\"Bearer {access_token}\"}\n",
+    "\n",
+    "time_zone_info = datetime.now(timezone.utc).astimezone().tzinfo\n",
+    "\n",
+    "def retrieve_s3_credentials():\n",
+    "    resp = requests.post(s3_credentials_url, headers=s3_cred_headers)\n",
+    "    s3_creds = resp.json().get(\"Credentials\")\n",
+    "    s3_creds_formatted = {\n",
+    "        \"access_key\": s3_creds.get(\"AccessKeyId\"),\n",
+    "        \"secret_key\": s3_creds.get(\"SecretAccessKey\"),\n",
+    "        \"token\": s3_creds.get(\"SessionToken\"),\n",
+    "        \"expiry_time\": datetime.fromtimestamp(s3_creds.get(\"Expiration\")).replace(tzinfo=time_zone_info).isoformat(),\n",
+    "    }\n",
+    "    print(\"Retrieved/refreshed s3 credentials\")\n",
+    "    return s3_creds_formatted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1fdb1fd2",
+   "metadata": {},
+   "source": [
+    "### Define callback method for logging upload progress"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 128,
+   "id": "45dac5b0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_progress_cb():\n",
+    "    lock = threading.Lock()\n",
+    "    uploaded_bytes = 0\n",
+    "    prev_percent = 0\n",
+    "\n",
+    "    def progress_cb(num_bytes):\n",
+    "        nonlocal uploaded_bytes\n",
+    "        nonlocal prev_percent\n",
+    "        should_update_progress_printout = False\n",
+    "        \n",
+    "        lock.acquire()\n",
+    "        uploaded_bytes += num_bytes\n",
+    "        percent_of_total_upload = float(\"{:.1f}\".format(uploaded_bytes / filesize * 100))\n",
+    "        if percent_of_total_upload > prev_percent:\n",
+    "            should_update_progress_printout = True\n",
+    "        prev_percent = percent_of_total_upload\n",
+    "        lock.release()\n",
+    "        \n",
+    "        if should_update_progress_printout:\n",
+    "            print(f\"{percent_of_total_upload}% uploaded\\r\", end=\"\")\n",
+    "            \n",
+    "    return progress_cb"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f648e0b2",
+   "metadata": {},
+   "source": [
+    "### Upload file using temporary s3 credentials"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4c332b6f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "session_creds = RefreshableCredentials.create_from_metadata(\n",
+    "    metadata=retrieve_s3_credentials(),\n",
+    "    refresh_using=retrieve_s3_credentials,\n",
+    "    method=\"sts-assume-role\",\n",
+    ")\n",
+    "session = get_session()\n",
+    "session._credentials = session_creds\n",
+    "boto3_session = boto3.Session(botocore_session=session)\n",
+    "s3 = boto3_session.client(\"s3\")\n",
```

Comment:
I got an `AttributeError` on this line: 🤔 
```
AttributeError: 'str' object has no attribute 'get'
```

Reply:
... which looked familiar, i.e. typically happens when running an AWS CLI command without specifying the `AWS_PROFILE`.

When I modified the line above to include the `profile_name`:
```
boto3_session = boto3.Session(botocore_session=session, profile_name="single-cell-dev")
```
It did work. 🤔

Reply:
Weird. I went and `unset AWS_PROFILE` in the window in which I was running the python notebook, I restarted the notebook, I did not otherwise set `AWS_PROFILE`, and it still worked for me.

Do you normally not have `AWS_PROFILE` set in your terminal?

Reply:
Hm, maybe the fact that I have a `default` entry in my `~/.aws/config` is getting the job done for me...

Reply:
@metakuni I'm not able to reproduce your error even after I move my aws config file. I wonder if we should move on to testing with curators, see if they get the same error.

Reply:
Yeah, I 'lucked out' since I don't have `AWS_PROFILE` exported in my env on my M1 laptop. 😄  Unfortunately, I didn't have a non-CZI laptop to try this on.

Reply:
Lucked out? Are you saying it works for you now?

I tried unsetting my `AWS_PROFILE` and I moved my config away from the expected location, and it still worked for me.